### ISSUE TITLE
Unreadable text due to colour choices

### ISSUE THREAD

**Issue by Kuffs2205** [luminous-software/start-page-plus#2]
### Installed product versions
- Visual Studio: 2019ENT
- This extension: 0.12.1

### Description
Colour choices are not in keeping with the UI theme and in some cases make the text  unreadable. See image here: https://www.screencast.com/t/t78aVgZPwyN

### Steps to recreate
1. Use a light theme 

### Current behavior
White text on a yellow background. Also, brown and yellow UI is unpleasant.

### Expected behavior
Contrasting text and use of colours from existing theme

**Comment by sevanmarc** [comment]
I get the same thing, but I thought it was because I use a heavily customised theme based around a blue (light) environment and dark editor which breaks a lot of stuff (including the new syntax highlighted find window in VS2019 :-( )

If you are using standard VS theme colours, a list of which ones you use where would be useful so I can tweak my theme to make it all work.

**Comment by yannduran** [comment]
HI guys! Thanks for taking the time to look at the _Start Page+_ preview, and to let me know about the color problem. Unfortunately I was aware of it, but forgot to have another look at it in my haste to get the preview (finally) out there.,

It's been a hard, if not impossible, task to try to pick brushes from the _Light_, _Dark_, and _Blue_ themes that work in **all three themes**. The brushes that I chose ARE from the themes' colors, it's just that it's been incredibly difficult to try to find color combinations that look good together in each one of the three themes. 

In the end I found I was spending so much time tweaking which brushes to use that I put it aside to get an actual working preview in the marketplace.

White text on a yellow background is just WRONG, I know. At one stage I had the _Blue_ theme looking good, but then I think the _Light_ theme didn't work, or some other combination. There was always one out of the three that didn't look that good.

The _Light_ theme currently looks like what a _Light_ theme user might recognise/expect. Same with the _Dark_ theme, but as you've seen, not the _Blue_ theme. Btw, are you using the _new blue theme_? Or the recreation of the _original blue theme_ from VS 2017?

The purple that you've probably seen in the images I've posted is from a _customised dark theme_ that I use myself. I needed something familiar to work with. If you're not used to seeing that horrid brown color, or the nice-but-not-with-white yellow, a work-around for the moment would be to change the values for those two colors in _Fonts and Colors_.

The brown should be from the _AccentDark_ key, the value used for my _ContentBackgroundBrush_ (used in the `StartView.xaml` file). The yellow should be from the _ToolWindowTabText_ key, the value used for my _ColumnBackgroundBrush_ (found in the `Styles.xaml` file). You should be able to change them to whatever you like, as it's unlikely that you'll ever see them elsewhere in VS (apparently).

If you, or anyone else reading this, would like to help out finding color combinations that work in all three themes I'm happy to accept that help. The code is all here in the GitHub repo. The Brushes.xaml     resource dictionary is all that needs to be changed theme-wise.

Thanks again!

**Comment by Kuffs2205** [comment]
Sorry. It's not the light theme. Its the light coloured ones such as the one I exported from VS2017 so I could have my original colours back.  I also have a bunch of others too that look similar such as "Green" and "Tan" but I think they came from the Theme Editor extension.  I'll see if I can tweak my theme accordingly.

**Comment by yannduran** [comment]
Yes, they have the colors listed one way in _Fonts and Colors_, and another way in the _theme editor_, and yet another way in `VsColors`. Even when I found a color I wanted to use, I often couldn't find it in `VsColors`.

**Comment by Kuffs2205** [comment]
Madness. I appreciate your issue.  
You could possibly avoid it by using your own static colours that look good and allow a setting to switch between dark and light versions (or if you can detect dark/light then even better).  Even the original start page doesn't really have much support for themes.

**Comment by yannduran** [comment]
If you's like to add an issue for that as a _feature request_, I'd be happy to look at it after preview status has ended, for sure.

**Comment by yannduran** [comment]
@sevenmark yes I am using _VS colors_, I'll try to get you a list as soon as i can, but if you look in the `brushes.xaml` file, you'll be able to see them for yourself if you don't mind digging into a code file (in case I take too long).

**Comment by sevanmarc** [comment]
@yannduran thanks for the info, when I get some time I'll dig into the file and pull the colours out. I'm happy to dig into xaml so I'm sure I can find what I need. I appreciate the problem, there are hundreds of colours in the theme and finding ones that work in all situations is pretty hard.

**Comment by yannduran** [comment]
@sevanmarc yes, the main problem is finding colors that work in all three main themes! :-)

**Comment by MagicAndre1981** [comment]
I also have this when using blue theme:

![image](https://user-images.githubusercontent.com/8274816/65505040-a683ed00-dec8-11e9-8fca-8402652c5029.png)

It is very hard to read.

**Comment by boris-df** [comment]
Same Problem; I'm using VSColor, i have "Tan" selected. 
Like you can see in Screenshot from MagicAndre1981 the Big titles (the not selected ones) are white on white)

with the Tipp from Kuffs2205 i changed "Tool Window>Tab>Unselected>Text" from white to nearly black to at least make the small text readable without making the VS UI unreadable ;)

I can imagine that this is a tricky part - but shouldn't it be possible to use Color-Constants from a Template to make it readable in (nearly) every case?

I mean: Why using "Tab>unselected>Text" for the  Path-Information? This could only work if at the same time the background is "Tab>unselected>Background" - which is not the case.

I don't have the time to search which background you are currently using :) but there should be a fitting constant for the Text available (i would prefer the same color as for the Project-name if it means that it is readable)

Same goes for the Headline - i think you are mixing color-constants from different groups... that would never work well i think

Just an idea - maybe i'm completly wrong :D

**Comment by Kuffs2205** [comment]
@boris-df Don't forget to change "Tool Window>Tab>Unselected>Background" to a contrasting colour also otherwise your tabs in the IDE will not be readable.

**Comment by boris-df** [comment]
> @boris-df Don't forget to change "Tool Window>Tab>Unselected>Background" to a contrasting colour also otherwise your tabs in the IDE will not be readable.

With "Tan" it's OK if you set the Text to nearly black. It works on the Tan Background in the IDE too. If i had more time i would look at the code to see which constants he is using and try to fix/change it - but that's not my No1 priority right now - too much work because of VS 16.3 + new Mono Update + XCode 11 + no Application Loader + .... #meh

**Comment by yannduran** [comment]
@boris-df 

> i think you are mixing color-constants from different groups

You're absolutely right! I originally started off selecting VsColors keys in the dark theme (actually a customised version of Dark) that gave me the look that I wanted and was horrified to see how they turned out in the other two major themes.

I wanted to use colors that users of each theme would find _familiar_ but also a bit _more_, so I went searching for color combinations that would accomplish that. Easier said than done unfortunately! 

Originally I managed to do that with both Dark and Blue, with Light not working out. Then when I reworked it I succeeded with Dark and Light, with Blue not working out. So I could only get two out of the three with acceptable results, due to the way Microsoft has structured the constants.

Eventually I realised that I was spending WAY too much time trying to tweak the colors (I did a LOT of tweaking of colors), and not having anything to be able to let people try out for themselves. I needed to get a preview out ASAP, so I put the colors on the back burner, and concentrated on the functionality.

**Comment by rogerfar** [comment]
The text colors are alot better, but it seems the background colors remained unchanged:
![image](https://user-images.githubusercontent.com/1142915/65897176-f3097580-e36b-11e9-869a-e7b47901dbfb.png)

**Comment by boris-df** [comment]
Hm, i'm sorry to say this but I don't think it's better ... but worse? :) 
See Screenshot from @rogerfar - The Main Text of an entry is now a dim grey. Looks like all projects are disabled now o.O 

I wish i had more time to take a look a deep look into the code to understand which entries of the Template are used... i still think that the logic/selection is not correct 
- Nowhere in my UI is the main text of any entry a dim grey but in this extension
- the Menu-Bar (Start, Favorit, etc.) is white on white for the not-selected-entries

i'm sorry that i only can tell what i don't like - i DO like the fact that there IS an extension like this :)

**Comment by yannduran** [comment]
@boris-df 

Don't worry, I agree with you, that it's _**_still_**_ not an **ideal** situation! But at least it's readable now. It's only a stop-gap measure until **I** can find the time to find a better solution to this problem. I do want ALL feedback though, good or bad.

A pull request was made by a helpful user to at least get it readable for the _Blue_ theme. I actually don't like how it looks in the _Dark_ theme now, which is what I use myself, but it is readable and I know it'll improve.

The main problem is actually a lack of consistency in Microsoft's themes, but you're also right about the choice of VsColor keys that I'm using. Believe me it wasn't just a quick choice. I spent hours and hours, going round and round in circles, trying to find values that gave me the overall effect that I wanted and looked OK in the 3 major themes.

Remember, this is still an early preview. It will get better. I just have to find more time.

**Comment by yannduran** [comment]
@rogerfar 

Don't worry, these are NOT the final colors. Blame Microsoft for the **_horrendous_** colors, and blame ME for not having found the right keys to use yet. Start Page+ is still in **early** preview, it will get better, I promise you!

**Comment by yannduran** [comment]
@Kuffs2205 
@sevanmarc 
@MagicAndre1981 
@boris-df 
@rogerfar 

Guys, I've just released a new preview (Preview 3) with just an updated color scheme (and a fixed link). 

I had an epiphany last night, triggered by something one of you said, but I can't find the comment now. Anyway I worked out what I hope will be a set of more theme-friendly colors.

You can see samples of various themes [on the marketplace][marketplace-url], or [on the website][website-url]. Please let me know what you think of them!

[marketplace-url]: https://marketplace.visualstudio.com/items?itemName=YannDuran.StartPagePlus
[website-url]: https://luminous-software.solutions/start-page-plus/#theme-samples

**Comment by Kuffs2205** [comment]
@yannduran way WAY better. The colour (color) no longer looks like the contents of a nappy (diaper)

**Comment by MagicAndre1981** [comment]
the blue looks fine on picture. I'll update the extension later in my 2019 test installation and see how it looks in real usage.

**Comment by yannduran** [comment]
Wow guys you were quick!!! Thanks for the feedback!

**Comment by sevanmarc** [comment]
Nice! 👍

**Comment by MagicAndre1981** [comment]
it is much better now with blue theme. Thanks for the improvement in preview 3

**Comment by yannduran** [comment]
I'm **really** glad you guys like it! 

If you haven't already done so, it'd be great if you could give the extension a rating, or a short review?

**Comment by MagicAndre1981** [comment]
I'll vote when the final is out
